compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateAIEWorkgroup.cpp

```diff
@@ -185,16 +185,19 @@ LogicalResult WorkgroupBuilder::buildForDmaCpyNdOp(
       circularDmaTargetOffsets, circularDmaTargetSizes,
       circularDmaTargetStrides, circularDmaSourceOffsets,
       circularDmaSourceSizes, circularDmaSourceStrides);
+  Type ty =
+      !sourceMemSpace
+          ? static_cast<Type>(
+                controlCodeRewriter.getType<AMDAIE::AsyncSourceTokenType>())
+          : static_cast<Type>(
+                controlCodeRewriter.getType<AMDAIE::AsyncTargetTokenType>());
   auto npuDmaCpy = controlCodeRewriter.createAndLookup<AMDAIE::NpuDmaCpyNdOp>(
```

Comment:
From what I infer in this part of the code - we add `async_source` for L3->L2 and `async_target` for other DMA transfers ? If that's too trivialized, please correct me.

Reply:
Yes